test/utils/client.dart

```diff
@@ -30,11 +31,23 @@ Future<void> deleteAllKeys() async {
   }
 }
 
-Future<void> setUpClient() async {
+Future<void> setUpClient({bool isHttp2 = false}) async {
```

Comment:
Can you set up some tests that actually use this new implementation?

Reply:
Also, can you write a small "how to" piece in the README? So the other users will know how to use it?